spinn_machine/tags/abstract_tag.py

```diff
@@ -29,34 +31,35 @@ class AbstractTag(object):
         "_port"
     ]
 
-    def __init__(self, board_address, tag, port):
+    def __init__(self, board_address: Optional[str], tag: int,
+                 port: Optional[int]) -> None:
```

Comment:
Again remove optionals here and see if it works - it probably does!

Reply:
board_address yes port harder

Reply:
port worked too